src/app/Shared/Services/api.types.ts

```diff
@@ -0,0 +1,576 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AlertVariant } from '@patternfly/react-core';
+import { Observable } from 'rxjs';
+
+export type ApiVersion = 'v1' | 'v2' | 'v2.1' | 'v2.2' | 'beta';
```

Comment:
This should have `v2.3` as well. There are no new `v2.4` API endpoints but it could be added here as well for consistency.